tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AdoNet/MySqlDatabaseMonitoringTests.cs

```diff
@@ -0,0 +1,120 @@
+// <copyright file="MySqlDatabaseMonitoringTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.AdoNet
+{
+    [Trait("RequiresDockerDependency", "true")]
+    public class MySqlDatabaseMonitoringTests : TracingIntegrationTest
+    {
+        public MySqlDatabaseMonitoringTests(ITestOutputHelper output)
+            : base("MySql", output)
+        {
+            SetServiceVersion("1.0.0");
+            SetEnvironmentVariable("DD_ENV", "testing");
+        }
+
+        public static IEnumerable<object[]> GetMySql8Data()
+        {
+            foreach (object[] item in PackageVersions.MySqlData)
+            {
+                if (!((string)item[0]).StartsWith("8") && !string.IsNullOrEmpty((string)item[0]))
+                {
+                    continue;
+                }
+
+                yield return item;
+            }
+        }
+
+        public override Result ValidateIntegrationSpan(MockSpan span) => span.IsMySql();
+
+        [SkippableTheory]
+        [MemberData(nameof(GetMySql8Data))]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitDbmCommentedSpanspropagationDisabled(string packageVersion)
+        {
+            SetEnvironmentVariable("DD_DBM_PROPAGATION_MODE", "disabled");
+            SubmitDbmCommentedSpans(packageVersion);
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(GetMySql8Data))]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitDbmCommentedSpanspropagationService(string packageVersion)
+        {
+            SetEnvironmentVariable("DD_DBM_PROPAGATION_MODE", "service");
+            SubmitDbmCommentedSpans(packageVersion);
+        }
+
+        [SkippableTheory]
+        [MemberData(nameof(GetMySql8Data))]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitDbmCommentedSpanspropagationFull(string packageVersion)
+        {
+            SetEnvironmentVariable("DD_DBM_PROPAGATION_MODE", "full");
+            SubmitDbmCommentedSpans(packageVersion);
+        }
+
+        // Check that the spans have been tagged after the comment was propagated
+        private void ValidatePresentDbmTag(IEnumerable<MockSpan> spans)
+        {
+            var propagationLevel = GetEnvironmentVariable("DD_DBM_PROPAGATION_MODE");
```

Comment:
nit: You could pass down the value of DD_DBM_PROPAGATION_MODE all the way down into this method since you're hard-coding it at the beginning of each test.